src/compressed_tensors/compressors/model_compressor.py

```diff
@@ -146,16 +159,29 @@ def from_pretrained_model(
     def parse_sparsity_config(compression_config: Dict) -> Union[Dict, None]:
         if compression_config is None:
             return None
+        if SPARSITY_CONFIG_NAME not in compression_config:
+            return None
+        if hasattr(compression_config, SPARSITY_CONFIG_NAME):
+            # for loaded HFQuantizer config
+            return getattr(compression_config, SPARSITY_CONFIG_NAME)
         return compression_config.get(SPARSITY_CONFIG_NAME, None)
 
     @staticmethod
     def parse_quantization_config(compression_config: Dict) -> Union[Dict, None]:
+        if compression_config is None:
+            return None
+        if QUANTIZATION_CONFIG_NAME not in compression_config:
+            return None
+
+        if hasattr(compression_config, QUANTIZATION_CONFIG_NAME):
+            # for loaded HFQuantizer config
+            return getattr(compression_config, QUANTIZATION_CONFIG_NAME)
```

Comment:
Doesn't this case not apply anymore? This would be checking for compression_config: {quantization_config: {}} but the quantization config is top level now

Reply:
right now, we nest the actual quantization config under it - this is a case for an instantiated config
https://github.com/neuralmagic/transformers/pull/101/files#diff-4333c40134efe7287ccda3bdd11e266b90a62629b933bf2cd15fa39cbf23b088R1074

we could define another pathway for loading since these cases are a bit roundabout for instantiating

Reply:
Ah ok I see, I had thought you were searching in the config.json dictionary. Makes sense!